### PR TITLE
hack: add verify-shellcheck.sh, address shellcheck failures

### PR DESCRIPTION
**Pull request by spiffxp**
With all the bash we have, I think it's time to start enforcing shellcheck failures. I managed to address a lot of them piecemeal with my PRs to infra/gcp and audit, because my editor has shellcheck integration, and I'm a completionist. So I decided to go the rest of the way and address the remaining failures.

As of this PR, verify-shellcheck.sh will run by default as part of pull-k8sio-verify

None of the failures were so annoying that I felt compelled to add a .shellcheck config file to exclude

See individual commits for the specific failures fixed. I have run all of these scripts in gcr.io/k8s-staging-infra-tools/k8s-infra:latest to verify lack of breakage

### CODE REVIEW

**Comment by spiffxp** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/k8s.io/2201/pull-k8sio-verify/1403406086211375104#1:build-log.txt%3A98 - proof that verify-shellcheck.sh is running and passing

**Comment by spiffxp** [comment]
/cc @ameukam @BenTheElder 

/assign @thockin 
FYI, doesn't require your attention, but thought you might want to know

**Comment by BenTheElder** [comment]
i think actually it was a bit of shell that verified that `*.sh` things were executuable, using `find` fyi https://github.com/kubernetes/test-infra/pull/11357

File: k8s.gcr.io/images/k8s-staging-csi-secrets-store/generate.sh [hunk 1]
```diff
@@ -14,12 +14,23 @@
...
     echo "  dmap:"
-    gcloud container images list-tags gcr.io/k8s-staging-csi-secrets-store/$repo --format='get(digest, tags)' --sort-by='tags' --filter='tags~^v[0-9]+.[0-9]+.[0-9]+$ AND NOT tags=v0.0.11' | sed -e 's/\([^ ]*\)\t\(.*\)/    "\1": [ "\2" ]/'
-done
+    gcloud container images list-tags \
+        "${repo}/$image" \
+        --format="get(digest, tags)" \
+        --sort-by="tags" \
+        --filter="${tag_filter}" \
+        | sed -e 's/\([^ ]*\)\t\(.*\)/    "\1": [ "\2" ]/'
+done
```
Comment (tam7t) [inline k8s.gcr.io/images/k8s-staging-csi-secrets-store/generate.sh:36]: nit: newline

File: k8s.gcr.io/images/k8s-staging-sig-storage/generate.sh [hunk 2]
```diff
@@ -14,30 +14,41 @@
...
-        sed -e 's/\([^ ]*\)\t\(.*\)/    "\1": [ "\2" ]/'
-done
-
+    gcloud container images list-tags \
+        "${repo}/$image" \
+        --format="get(digest, tags)" \
+        --sort-by="tags" \
+        --filter="${tag_filter}" \
+        | sed -e 's/\([^ ]*\)\t\(.*\)/    "\1": [ "\2" ]/'
+done
```
Comment (tam7t) [inline k8s.gcr.io/images/k8s-staging-sig-storage/generate.sh:54]: nit: newline

File: dns/lib.sh [hunk 3]
```diff
@@ -31,7 +31,7 @@ precook_zone_configs() {
     for z in "${zones[@]}"; do
         # Every zone should have 1 file $z.yaml or N files $z._*.yaml.
         # $z already ends in a period.
-        cat zone-configs/${z}yaml zone-configs/${z}_*.yaml \
+        cat "zone-configs/${z}"yaml "zone-configs/${z}"_*.yaml \
```
Comment (thockin) [inline dns/lib.sh:34]: This looks like it is missing a period?

**Pull request merged**